### PR TITLE
[sc-2651] Connector for Amazon S3

### PR DESCRIPTION
**Pull request by operramon**
Add AWS S3 source connector to desktp app

### CODE REVIEW

File: apps/desktop/src/app/sync/models.ts [hunk 1]
```diff
@@ -19,8 +19,9 @@ export interface ISourceConnector {
   hasServerSideAuth: boolean;
   resumable: boolean;
   getParameters(): Observable<Field[]>;
+  handleParameters?(params: ConnectorParameters): void;
```
Comment (mpellerin42) [inline apps/desktop/src/app/sync/models.ts:22]: shouldn't the `?` be just before the `:` ?
```handleParameters(params: ConnectorParameters)?: void;```

File: apps/desktop/src/app/sync/sources/s3.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        type: 'text',
+        required: true,
+      },
+      {
+        id: 'secret_access_key',
+        label: 'Secret access key',
+        type: 'text',
+        required: true,
+      },
+    ]);
```
Comment (mpellerin42) [inline apps/desktop/src/app/sync/sources/s3.ts:61]: should we translate the labels from this list?

File: apps/desktop/src/app/sync/sources/s3.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  private filterResult(result: _Object, query?: string) {
+    if (query && result.Key) {
+      const regex = new RegExp(`(${query})`, 'i');
+      return regex.test(result.Key);
+    }
+    return true;
+  }
+
+  download(resource: SyncItem): Observable<Blob> {
+    const command = new GetObjectCommand({
```
Comment (ebrehault) [inline apps/desktop/src/app/sync/sources/s3.ts:132]: Actually, the objective with Amazon S3 (and with Google Cloud Service) is not to download the physical and then upload it back to Nuclia like we did with DropBox or GDrive.
We want to use them as cloud file storage, so we want to use the Nuclia cloud file support, meaning we will create a FileField by providing its direct URL to Amazon S3 (and also some extra_headers to provide an auth token or whatever Amazon S3 in order to give access to the file)

File: apps/desktop/src/app/sync/models.ts (hunk 1 above)
Comment (operramon) [reply apps/desktop/src/app/sync/models.ts:24]: If `?` is at the end then Typescript complains, I don't know why.

File: apps/desktop/src/app/sync/sources/s3.ts (hunk 2 above)
Comment (operramon) [reply apps/desktop/src/app/sync/sources/s3.ts]: I added translations for `Region`.
Not sure if `Access key id` and 'Secret access key' should be translated or not.
In the case of `Bucket` it's the same for all langs.

**Pull request merged**